packages/changed-elements-react/src/NamedVersionSelector/NamedVersionSelector.tsx

```diff
@@ -127,6 +138,7 @@ export function NamedVersionSelectorWidget(props: NamedVersionSelectorWidgetProp
               iModelConnection={iModel}
               manager={manager}
               usingExperimentalSelector
+              onLifeCycleChange={handleLifecycleChange}
```

Comment:
I'm not a fan of rippling the lifecycle events upward, seems like an antipattern in react world. Why not simply leverage the comparison events to decide whether or not to show the different parts of the widget here? The real issue is that we were allowing going back to the main widget entry before stopping comparison correct? Why not simply act on the appropriate manager event (e.g. versionCompareStopped, versionCompareStarting, etc.)